src/core/mac/sub_mac_csl_receiver.cpp

```diff
@@ -118,17 +118,26 @@ bool SubMac::UpdateCsl(uint16_t aPeriod, uint8_t aChannel, ShortAddress aShortAd
     mCslChannel = aChannel;
 
     VerifyOrExit(diffPeriod || diffPeer);
+
     mCslPeriod    = aPeriod;
     mCslPeerShort = aShortAddr;
     IgnoreError(Get<Radio>().EnableCsl(aPeriod, aShortAddr, aExtAddr));
 
     mCslTimer.Stop();
-    if (mCslPeriod > 0)
+    if (aPeriod > 0)
     {
         mCslSampleTime = TimeMicro(static_cast<uint32_t>(Get<Radio>().GetNow()));
         mIsCslSampling = false;
+        mCslPeriod     = aPeriod;
         HandleCslTimer();
     }
+    else if ((mState == kStateRadioSample) && (!RadioSupportsReceiveTiming()))
+    {
+        // Give WED a chance to enter receive or sleep state, and give SubMac a chance to enter sleep state.
+        RequestSleep(kRequesterCsl);
```

Comment:
The comment said "Give WED a chance to enter receive or sleep state". Why it's not `RequestSleep(kRequesterWed);`?

Reply:
Calling `RequestSleep()` is to give up control of the radio. If no module is using the radio, the radio should enter the sleep state. Otherwise, the radio could be controlled by the other module.

Here, the CSL is disabled, and it won't control the radio anymore. `mState == kStateRadioSample` means the Mac layer has called `Sleep()` to request the SubMac to enter the sleep state.  Here, CSL calls the `RequestSleep()`.  If the WED is controlling the radio (WED is enabled), `RequestSleep()` lets the WED to control the radio. Otherwise, `RequestSleep()`  sets the radio to sleep state (Mac layer's request).

Reply:
Understood, thanks!